### PR TITLE
Input parse

### PR DESCRIPTION
**Pull request by FlanFlanagan**

### CODE REVIEW

**Comment by FlanFlanagan** [comment]
hold up. This is not done.

File: rickshaw/generate.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 from collections.abc import Sequence
 from copy import deepcopy
 from random import randrange, choice
+from rickshaw import simspec
```
Comment (scopatz) [inline rickshaw/generate.py]: PEP8 this should be with the rickshaw imports below

File: rickshaw/generate.py [hunk 2]
```diff
@@ -591,6 +429,7 @@ def generate_region_inst(sim):
         }
     entries = sim["region"]["institution"]["initialfacilitylist"]["entry"]
     for facility in sim["facility"]:
+        print(facility)
```
Comment (scopatz) [inline rickshaw/generate.py]: Is this needed?

File: rickshaw/main.py [hunk 3]
```diff
@@ -20,25 +20,24 @@ def main(args=None):
     p.add_argument('-rh', dest='rh', action="store_true", help='runs the simulations after they have been generated')
     p.add_argument('-v', dest='v', action="store_true", help='verbose mode will pretty print generated files')
     ns = p.parse_args(args=args)
-    simspec = {}
+    temp_spec = {}
```
Comment (scopatz) [inline rickshaw/main.py]: non-blocking: `temp_spec` is not a great variable name because it isn't super descriptive.

Comment (scopatz) [reply rickshaw/main.py]: Just call it spec or something

File: rickshaw/server_scheduler.py [hunk 4]
```diff
@@ -54,7 +54,9 @@ def __del__(self):
     def start_cyclus_server(self):
         """Starts up a cyclus server at a remote location."""
         print("starting cyclus server")
-        cc = self.cyclus_service = self.client.services.create(self.server_tag)
+        cc = self.cyclus_service = self.client.services.create(self.server_tag,
+                                                               hostname = "",
+                                                               mode = ServiceMode('global'))
```
Comment (scopatz) [inline rickshaw/server_scheduler.py]: PEP8 no spaces around `=` in function calls

File: rickshaw/simspec.py [hunk 5]
```diff
@@ -0,0 +1,294 @@
+import collections
```
Comment (scopatz) [inline rickshaw/simspec.py:6]: Needs a module-level docstring and API auto-doc hooks

File: rickshaw/simspec.py [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+    special_calls : dict
+        Returns callable function associated with a Cycamore archetype and 
+        function identifier  
+    default_sources : set
+        Contains defaults for simulation sources. Not yet customizable by user.
+    default_sinks : set
+        Contains defaults for simulation sinks. Not yet customizable by user.
+    annotations : dict
+        Container for archetype annotations. 
+    """
```
Comment (scopatz) [inline rickshaw/simspec.py:250]: The above args are not actually accepted by the init function below

**Comment by FlanFlanagan** [comment]
ping @scopatz

**Comment by scopatz** [comment]
Looks good to me.  Can you please add some top-level docs to the docs directory?

**Comment by scopatz** [comment]
i.e. docs on how to use this, thanks!

**Comment by FlanFlanagan** [comment]
I can, but I'm going to merge this in and make that an issue. This is required for the input stuff I gave to yarden.

**Comment by scopatz** [comment]
So practical!

**Pull request merged**